include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -415,10 +415,12 @@ struct __histogram_general_private_global_atomics_submitter<__internal::__option
         using _bin_type = oneapi::dpl::__internal::__value_t<_Range2>;
         using _histogram_index_type = ::std::int32_t;
 
-        auto __global_mem_size = __q.get_device().template get_info<sycl::info::device::global_mem_size>();
-        const ::std::size_t __max_segments =
-            ::std::min(__global_mem_size / (__num_bins * sizeof(_bin_type)),
-                       oneapi::dpl::__internal::__dpl_ceiling_div(__n, __work_group_size * __min_iters_per_work_item));
+        const std::uint64_t __global_mem_size = __q.get_device().get_info<sycl::info::device::global_mem_size>();
+        const auto __max_groups =
+            oneapi::dpl::__internal::__dpl_ceiling_div(__n, __work_group_size * __min_iters_per_work_item);
```

Comment:
It seems like we should just use `std::uint64_t` instead of `auto` here and replace the common type thing in the next line with just `std::uint64_t`.
Is there a reason not to do that?

Reply:
That would be even better, thanks. std::uint64_t is large enough to accommodate any value in this context.